docs/dev/userland.md

```diff
@@ -18,70 +18,180 @@ All Rights Reserved. (Contributor contact(s):________________[Insert hyperlink/a
+| `build` | configure and build |
+| `install` | install software into the prototype directory |
+| `sample-manifest` | generate a sample IPS manifest based on the files installed to the prototype directory |
+| `publish` | publish the package to the local repository |
+| `pre-publish` | run all pre-publication checks (does actually what publish does, just without sending package to local repository) |
+| `REQUIRED_PACKAGES` | guess and generate build dependencies for the packages, manual edit might be needed |
+| `env-check` | check build environment for missing packages |
+| `env-prep` | install missing build dependencies (requires elevated privileges) |
+
+<i class="fa fa-info-circle fa-lg" aria-hidden="true"></i> **NOTE:**
+<div class="well">
+<p><b>Before adding new packages to oi-userland...</b></p>
+Before considering adding a new package to oi-userland, please check first whether someone else is working on the package by checking the issue tracker, mailing <a href="mailto:oi-dev@openindiana.org">oi-dev@openindiana.org</a> or asking on the IRC (#oi-dev at irc.freenode.net)
+<ul>
+<li>If you don't find anyone already working on a port, please register your effort by opening an issue.</li>
+<li>If you wish to update an existing port, look at the log for the component Makefile ("git log Makefile") and make sure you either contact the person who last updated the Makefile or include them on notifications for the issue by ticking their name.</li>
+</ul>
+This will ensure efforts aren't duplicated and help to ensure sanity and comity amongst project members.
+</div>
+
+### Setting up your build environment
+
+We strongly recommend building packages inside a fresh local zone set up exclusively for building. See [Quick zone setup example](../handbook/systems-administration/#quick-setup-example) for simple instructions.
+
+Further we assume that you are logged into the build zone if you set up the build environment in a zone, the directory were oi-userland is cloned can otherwise be anywhere you like.
+
+#### Adding RBAC profile to the build user
+
+Installing software requires privileges, so your build user must have at minimum the 'Software Installation' profile:
 
 ```bash
-git clone https://github.com/OpenIndiana/oi-userland.git
-cd oi-userland
+$ profiles
+Software Installation
+ZFS File System Management
+Console User
+Suspend To RAM
+Suspend To Disk
+Brightness
+CPU Power Management
+Network Autoconf User
+Desktop Removable Media User
+Basic Solaris User
+All
+```
+
+If it is not the case, add this profile to your build user:
+
+```bash
+pfexec su -
+usermod -P'Software Installation' <username>
 ```
 
-### Add your Github repository as remote:
+This is not necessary if your user has already the 'Primary Administrator' profile.
+
+#### Downloading oi-userland
+
+Start by [forking](https://github.com/OpenIndiana/oi-userland/fork) oi-userland repository on Github and then check out the repository (subdirectory oi-userland must not pre-exist):
```

Comment:
I'd be cautious to use the `/fork` link directly, better just point to the project and let user do the forking on their own.